settings.gradle

```diff
@@ -31,7 +27,7 @@ buildCache {
         region = 'fra1'
         bucket = 'testcontainers'
         path = 'cache'
-        push = isMasterBuild
+        push = !(System.getenv("READ_ONLY_REMOTE_GRADLE_CACHE") || !isMasterBuild)
```

Comment:
```suggestion
        push = !(System.getenv("READ_ONLY_GRADLE_CACHE") || !isMasterBuild)
```
?

Reply:
Also, can we invert boolean logic? :sweat_smile: 
`push = !System.getenv("READ_ONLY_GRADLE_CACHE") && isMasterBuild`

Reply:
Also, are we pushing Windows builds like this and do we want to push Windows builds?

Reply:
It is `READ_ONLY_REMOTE_GRADLE_CACHE` because local Gradle cache can still be enabled

Re inverting the logic - I find "push if NOT(READ_ONLY OR NOT(MASTER_BUILD))" easier to understand since you can focus on what's inside NOT :) @rnorth WDYT?

Re Windows builds - yes, why not?

P.S. consider starting new comments (you can do multiple on the same line) to avoid "RE"s ;)

Reply:
TBH I thought `READ_ONLY_REMOTE_GRADLE_CACHE` is our own environment variable, and you were setting `READ_ONLY_GRADLE_CACHE` in CI, therefore I thought this is just a typo.

Reply:
it is our own variable. Did I set it to `READ_ONLY_GRADLE_CACHE`? :D good catch then, I should change! 👍

Reply:
Resolve with regards to ENV name :slightly_smiling_face: 
About the boolean expression, I don't want to block just different taste, so proceed as you deem fit :man_shrugging: